### PR TITLE
Convert setup files to shellscript

### PR DESCRIPTION
**Pull request by aslafy-z**
As they only run shell commands.

### CODE REVIEW

**Comment by jasbur** [comment]
Great contribution. The `initial-setup.py` script orginially had a lot more logic (that proved to be unnecessary) in it. It didn't even occur to me that only shell commands were left until you submitted this pull request.

I looked over all of the code changes and everything looks good. I'm just going to test it locally when I get the chance and I'll merge it into the master branch after that.

**Comment by jasbur** [comment]
I had a couple of issues while testing your code:

- There should be a `clear` command on initially running the script to clear the screen of all previous output before running. The same `clear` command should probably be issued after each system command the script executes to keep the screen from getting too cluttered.

- The final prompt before executing the setup should probably be a little bit more verbose. It currently says "Confirm setup [y/N]:". Maybe it could say something like "Changes about to be committed to your Raspberry Pi. Continue? [y/N]"

- The `auto_config=0` line in `/etc/raspiwifi/raspiwifi.conf` doesn't get changed regardless of what's entered during the initial_setup.sh script.

- There seems to be a lot of extra console output generated while the script is running that looks like it should be hidden from the user. I'll attach a screenshot below:

console output

all of those lines starting with `+` should really be hidden from the user.

**Comment by aslafy-z** [comment]
I fixed the few points you highlighted. 

I did not added the clear commands between steps as I dont like scripts to clear my screen without asking, I want to be able to scroll back up to see the whole script output.
Are you sure you want me to add them ? I think it's kind of a anti-pattern for building script ran by users. What do you think about that ?

**Comment by jasbur** [comment]
Good point about clearing the screen. Is there any way we could make a better visual delineation to break up the output a bit? Something like a cosmetic header for each section (updating Apt, installing prerequisites, installing Flask, etc.)?

Also, the other changes look good to me. I'll test them as soon as I get the chance.

**Comment by aslafy-z** [comment]
In order to avoid code repetition I did a little function that prints a "header".
It could be enhanced but there is the first version of it:

```
print_header() {
  c='='
  line=$(head -c ${#1} < /dev/zero | tr '\0' $c)
  cat <<EOK
$c$c$c$line$c$c$c
$c$c $1 $c$c
$c$c$c$line$c$c$c
EOK
}
```

You can try it by copying pasting to your terminal and then call

```
print_header "I'm a category"
```

You'll end up with 

```
====================
== I'm a category ==
====================
```

I can add it between each install steps if it fits you

File: setup_lib.sh [hunk 1]
```diff
@@ -27,6 +27,6 @@ copy_config() {
 
 update_config() {
   sed -i "s/RaspiWiFi Setup/${CONFIG_SSID:-RaspiWiFi Setup}/" /etc/raspiwifi/raspiwifi.conf
-  [[ $AUTO_CONFIG = "y" ]] && sed -i "s/auto_config=0/auto_config=1/" /etc/raspiwifi/raspiwifi.conf
+  [ "x$AUTO_CONFIG" = "xy" ] && sed -i "s/auto_config=0/auto_config=1/" /etc/raspiwifi/raspiwifi.conf
```
Comment (jasbur) [inline setup_lib.sh]: Just curious, why is the `x` prepended on the answer here? Why resolve to `xy` instead of just `y`?

**Comment by jasbur** [comment]
All of the code seems to be working now. I like the header code you added in the comment above too. Do you want to implement it before I merge your Pull Request or do you want me to take care of it after?

File: setup_lib.sh (hunk 1 above)
Comment (aslafy-z) [reply setup_lib.sh]: I made this to take care of edge cases with some versions of bash, when the tested variable is empty, more details here: https://stackoverflow.com/questions/174119/why-do-shell-script-comparisons-often-use-xvar-xyes

Comment (jasbur) [reply setup_lib.sh]: Ah, that makes perfect sense. Thanks.

**Comment by aslafy-z** [comment]
I just pushed the changes. Be free to merge when you're able to try it.

**Comment by jasbur** [comment]
I just ran the new code and had a failure right after Flask was installed:

capture

**Comment by aslafy-z** [comment]
Strange behavior. I fixed few things, but I don't know what happened. 
I added a way to dry-run it (`./initial_setup.sh test`)

**Comment by jasbur** [comment]
I haven't had a chance to try your latest code yet, but I'm a little worried that the benefits of switching to a Shell script might be getting diluted a bit. We actually now have a fair amount of logic happening during execution and I feel that Python might be a more logical choice of language. Using Python also keeps the code base more homogeneous throughout.

I do like the new features you've added, such as the dynamic headers, sudo check, and dry run options though. Maybe those could be implemented with Python?

**Pull request closed without merging**